website/pages/docs/k8s/upgrade/index.mdx

```diff
@@ -7,63 +7,303 @@ description: Upgrade Consul on Kubernetes
+node running that replica, it will be briefly deregistered from the mesh. Since
+it's the only replica, other services will not be able to make calls to that
+service. (NOTE: This can also be avoided by stopping that replica so it is rescheduled to
+a node whose Consul client has already been updated.)
+
+Given the above, we recommend that after Consul servers are upgraded, the Consul
+client daemonset is set to use the `OnDelete` update strategy and Consul clients
+are deleted one by one or in batches. See below for details.
 
 ## Upgrading Consul Servers
 
-To initiate the upgrade, change the `server.image` value to the
-desired Consul version. For illustrative purposes, the example below will
-use `consul:123.456`. Also, set the `server.updatePartition` value
-_equal to the number of server replicas_:
+To initiate the upgrade:
+
+1. Change the `global.image` value to the desired Consul version
+1. Set the `server.updatePartition` value _equal to the number of server replicas_:
+1. Set the `updateStrategy` for clients to `OnDelete`.
 
 ```yaml
-server:
+global:
   image: 'consul:123.456'
-  replicas: 3
+server:
   updatePartition: 3
+client:
+  updateStrategy: |
+    type: OnDelete
 ```
 
 The `updatePartition` value controls how many instances of the server
 cluster are updated. Only instances with an index _greater than_ the
 `updatePartition` value are updated (zero-indexed). Therefore, by setting
 it equal to replicas, none should update yet.
 
-Next, run the upgrade. You should run this with `--dry-run` first to verify
-the changes that will be sent to the Kubernetes cluster.
+The `updateStrategy` controls how Kubernetes rolls out changes to the client daemonset.
+By setting it to `OnDelete`, no clients will be restarted until their pods are deleted.
+Without this, they would be redeployed alongside the servers because their Docker
+image versions have changed. This is not desirable because we want the Consul
+servers to be upgraded _before_ the clients.
 
-```shell-session
-$ helm upgrade consul ./
-...
+Next, run the upgrade.
+
+```bash
+helm upgrade consul hashicorp/consul --version <your-version> -f /path/to/my/values.yaml
 ```
 
-This should cause no changes (although the resource will be updated). If
+This will not cause the servers to redeploy (although the resource will be updated). If
 everything is stable, begin by decreasing the `updatePartition` value by one,
-and running `helm upgrade` again. This should cause the first Consul server
+and running `helm upgrade` again. This will cause the first Consul server
 to be stopped and restarted with the new image.
 
-Wait until the Consul server cluster is healthy again (30s to a few minutes)
-then decrease `updatePartition` and upgrade again. Continue until
+Wait until the Consul server cluster is healthy again (30s to a few minutes).
+This can be confirmed by running `consul members` on one of the previous servers
+and ensuring that all three servers are listed and are `alive`.
```

Comment:
small nit: we've removed `replicas:3` above so `three` here is not always correct based on user's configurations.

```suggestion
and ensuring that all servers are listed and are `alive`.
```